components/dialogs/AuthDialog.vue

```diff
@@ -318,25 +330,86 @@ export default {
       switch (platform) {
         case 'email':
           this.currentTab = 'email';
-          break;
+          return;
 
         case 'wallet':
           this.currentTab = 'loading';
           this.startWeb3AndCb(this.signInWithWallet);
-          break;
+          return;
 
-        default: {
+        case 'google':
+        case 'twitter':
           this.signInPayload = await firebasePlatformSignIn(platform);
-          this.login();
-        }
+          break;
+
+        case 'facebook':
+          try {
+            this.currentTab = 'loading';
+            this.signInPayload = await new Promise((resolve, reject) => {
+              if (!window.FB) {
+                reject(new Error('FACEBOOK_SDK_NOT_FOUND'));
+              }
+              window.FB.login(({ authResponse }) => {
+                if (authResponse && authResponse.accessToken) {
+                  const {
+                    accessToken,
+                    userID: platformUserId,
+                  } = authResponse;
+
+                  resolve({
+                    accessToken,
+                    platformUserId,
+                  });
+                } else {
+                  reject(new Error('FACEBOOK_AUTH_REJECTED'));
+                }
+              }, { scope: 'public_profile,pages_show_list,user_link' });
+            });
+          } catch (err) {
+            this.currentTab = 'portal';
+          }
+          break;
+
+        default:
       }
+
+      this.login();
     },
     async signInWithEmail(email) {
       this.email = email;
       this.currentTab = 'loading';
       await firebaseSendSignInEmail(email);
       this.currentTab = 'checkInbox';
     },
+    async signInWithWallet() {
+      this.currentTab = 'signingIn';
+
+      // Determine whether the wallet has registered
+      try {
+        await apiCheckIsUser(this.getLocalWallet);
+      } catch (err) {
+        // Assume it is 404
+        // Redirect user to register
+        this.signInPayload = {
+          wallet: this.getLocalWallet,
+        };
+        this.currentTab = 'register';
+        return;
+      }
+
+      try {
+        this.signInPayload = await User.signLogin(this.getLocalWallet);
+        this.login();
+      } catch (err) {
+        if (err.message.indexOf('Invalid "from" address') >= 0) {
```

Comment:
TODO: should be guarded by actually retrieving the wallet address before asking user to sign

Reply:
@mckingho please look into why we are not polling address change later